### ISSUE TITLE
Fix S4275 FP: Inherited properties

### ISSUE THREAD

**Issue by antonioaversa** [SonarSource/sonar-dotnet#8137]
Originated from https://github.com/SonarSource/sonar-dotnet/issues/8110

The analyzers shouldn't raise an issue when:
- `get` and `set` override a `virtual` property accessing a `private` field of the base class
- `get` and `set` of the derived class call `base` behaviors, and don't directly access a field

Minimal example:
```cs
public class Base
{
    private bool _isSelected;

    public virtual bool IsSelected
    {
        get => _isSelected;
        set => _isSelected = value;
    }
}

public class Derived : Base
{
    public override bool IsSelected
    {
        get => base.IsSelected;
        set => base.IsSelected = value;
    }
}
```

The issue shouldn't be raised because the derived class cannot access the `private` field of the base class.
Moreover, even if it could (e.g. the field of the base class is `protected`), it likely shouldn't, and it should delegate to the base class, calling `base` behaviors instead.

**Comment by antonioaversa** [comment]
Closed as duplicate of https://github.com/SonarSource/sonar-dotnet/issues/8110
